app/services/notification_service.py

```
@@ -242,6 +242,15 @@ def _format_event(self, event_type: NotificationEventType, event_data: EventData
                 f"IMMEDIATE ACTION REQUIRED!"
             )
 
+        if isinstance(event_data, StoragePermissionErrorData):
```

Ruff (0.15.9)

[error] Undefined name `StoragePermissionErrorData`

(F821)